WordPress/gutenberg
# Adds suggestions for categories and formats to LinkControl

## Description
Closes #20924 and closes #20165 

## How has this been tested?
Tested locally by:

0. Make sure you are:
a) having some categories setup
b) on a theme that supports post formats (twentyfourteen)
c) some posts saved as a post formats other than standard
1. Add a new post
2. Add a navigation block in that post
3. Add a new navigation link
4. Search for something that is included in a category's or a post format's name
5. Observe the category and/or post format is included in suggestions

## Types of changes
- adds a format search handler
- adds a category search handler
- hooks the new handlers in the search endpoint
- updates LinkControl's fetchLinkSuggestions to handle formats and categories as well

**PS**
I don't know if this is the best way to implement this feature. However, because we need the link to the format index page, I can't see a better way to get it than from an API call. We could output formats in the bulk data which is rendered on load, but categories could be too many for this option. 

cc @getdave - this updates indirectly `LinkControl` to show more types of suggestions. Suggestions welcome! :)
